test/e2e_limit_classes/limitClasses/ingressClasses.go

```diff
@@ -66,15 +66,28 @@ var _ = ginkgo.Describe("Test limitclass on fromHost", ginkgo.Ordered, func() {
 
 	ginkgo.It("should only sync ingressClasses with allowed label to vcluster", func() {
 		ginkgo.By("Listing all ingresssesClasses available in vcluster")
-		ics, err := f.VClusterClient.NetworkingV1().IngressClasses().List(f.Context, metav1.ListOptions{})
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
-		var names []string
-		for _, ic := range ics.Items {
-			names = append(names, ic.Name)
-		}
-		gomega.Expect(names).To(gomega.ContainElement(nginxClassName))
+		gomega.Eventually(func() []string {
+			ics, err := f.VClusterClient.NetworkingV1().IngressClasses().List(f.Context, metav1.ListOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			var names []string
+			for _, ic := range ics.Items {
+				names = append(names, ic.Name)
+			}
+			return names
+		}).WithPolling(time.Second).WithTimeout(framework.PollTimeout).Should(gomega.ContainElement(nginxClassName))
+
 		ginkgo.By("Found nginx-ingressclass in vcluster")
```

Comment:
Can we remove these statements after the check has been performed, and if needed, write this in the eventually block when failure happens.